wallet/wallet-cli/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::VecDeque, path::PathBuf, sync::Arc};
+
+use clap::Parser;
+use common::chain::ChainConfig;
+use dialoguer::{theme::ColorfulTheme, History};
+use node_comm::{make_rpc_client, node_traits::NodeInterface};
+use utils::default_data_dir::PrepareDataDirError;
+
+use crate::config::Commands;
+
+mod config;
+
+const MAX_HISTORY: usize = 1000;
+
+#[derive(thiserror::Error, Debug)]
+pub enum WalletCliError {
+    #[error("RPC error: {0}")]
+    RpcError(node_comm::rpc_client::NodeRpcError),
+    #[error("Wallet error: {0}")]
+    WalletError(wallet::wallet::WalletError),
+    #[error("Console IO error: {0}")]
+    ConsoleIoError(std::io::Error),
+    #[error("Cookie file {0} read error: {1}")]
+    CookieFileReadError(PathBuf, std::io::Error),
```

Comment:
When I run the CLI, I get this error in a panic form:

```
Error: CookieFileReadError("/home/samer/.mintlayer/mainnet/.cookie", Os { code: 2, kind: NotFound, message: "No such file or directory" })
```

which is reasonable from a technical point of view. But we should have a nicer message to the user. Consider a user who doesn't really know what a cookie is or how that works. We should have a way to detect whether the node is running, and not attempt connection (or anything related) unless it exists.

Reply:
It looks like this now:

wallet-cli launch failed: RPC authentication cookie-file /home/username/.mintlayer/regtest/.cookie read error: No such file or directory (os error 2). Please make sure the node is started.

Reply:
Alright. I can live with that for now.